espnet2/bin/asr_inference.py

```diff
@@ -470,13 +487,19 @@ def inference(
             assert len(keys) == _bs, f"{len(keys)} != {_bs}"
             batch = {k: v[0] for k, v in batch.items() if not k.endswith("_lengths")}
 
-            # N-best list of (text, token, token_int, hyp_object)
+            # log number of 10ms input frames needed for RTF calculations
```

Comment:
Why did you fix 10ms?
It can be changed depending on the hop size.

Reply:
Yeah, this is not nice. I fixed the logging message to report the input lengths in terms of number of 10ms frames because I think the already existing calculate_rtf.py assumes a frame shift of 10ms. Let me explain below in more detail.

In https://github.com/espnet/espnet/blob/master/utils/calculate_rtf.py#L51 the audio duration in seconds is calculated like this:
```
audio_sec += sum(audio_durations) / 100  # [sec]
```
I understood that audio_durations is actually the accumulated number of input frames as retrieved from the logfile:
```
if "INFO: input lengths" in x:
   audio_durations += [int(x.split("input lengths: ")[1])]
```
So for the above conversion of audio_durations to audio_sec I figured this implicitly assumes a 10ms frame shift:
```
sum(audio_durations) * 10 / 1000  # [sec] = sum(audio_durations) / 100  # [sec]
```
Is my thinking correct or did I miss something here?

Maybe a better solution could be to log the correct number of input frames (based on model's hop_size) and supply a frame_shift/hop_size argument to calculate_rtf.py?

Reply:
> Is my thinking correct or did I miss something here?

You're right!

Yeah, espnet1 fixed to use 10ms while espnet2 can change it (and many recipes are also using 8ms, now)...

> Maybe a better solution could be to log the correct number of input frames (based on model's hop_size) and supply a frame_shift/hop_size argument to calculate_rtf.py?

Sounds good.